### PR TITLE
[HMA-6139] Editable List View

### PR DESCRIPTION
**Pull request by tendai-moffatt**

### CODE REVIEW

File: Sources/SUIComponents/Organisms/EditableListView.swift [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            VStack(alignment: .leading, spacing: 0) {
+                Text(model.title, style: .H5)
+                    .padding(.spacer16)
+                ForEach(model.views.indices, id: \.self) { index in
+                    Components.Atoms.Stack(
+                        axis: EditableListView.isVertical ? .vertical(alignment: .leading) : .horizontal(alignment: .center),
+                        spacing: 0) {
+                            model.views[index]
+                                .typeErased
+                                .frame(minHeight: 21) // Minimum height of an Edit button
```
Comment (sweetclimusic) [inline Sources/SUIComponents/Organisms/EditableListView.swift:80]: Could this '21'  be a library constant or a property instead?

File: Sources/SUIComponents/Helpers/Extensions/View+HMRC.swift [hunk 2]
```diff
@@ -34,4 +34,19 @@ extension View {
...
     public var typeErased: AnyView { AnyView(self) }
+
+
+    /// Applies the given transform if the given condition evaluates to `true`.
+    /// - Parameters:
+    ///   - condition: The condition to evaluate.
+    ///   - transform: The transform to apply to the source `View`.
+    /// - Returns: Either the original `View` or the modified `View` if the condition is `true`.
+    /// https://www.avanderlee.com/swiftui/conditional-view-modifier/
+    @ViewBuilder func `if`<Content: View>(_ condition: @autoclosure () -> Bool, transform: (Self) -> Content) -> some View {
```
Comment (sweetclimusic) [inline Sources/SUIComponents/Helpers/Extensions/View+HMRC.swift:45]: Where is this used? in the EditableListView when testing the @State is Editable?

File: Sources/SUIComponents/Organisms/EditableListView.swift (hunk 1 above)
Comment (tendai-moffatt) [reply Sources/SUIComponents/Organisms/EditableListView.swift]: Updated in [9369659](https://github.com/hmrc/ios-swiftui-components/pull/33/commits/9369659d6dd224858f18586187308e9890226961)

**Review by sweetclimusic — APPROVED** [review]

File: Sources/SUIComponents/Organisms/EditableListView.swift [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+extension Components.Organisms {
+    public struct EditableListView2<Content: View>: View {
```
Comment (paulb00th) [inline Sources/SUIComponents/Organisms/EditableListView.swift:20]: Is EditableListView2 the best name for this?

**Review by paulb00th — CHANGES_REQUESTED** [review]
Looks good. One question though.

File: Sources/SUIComponents/Organisms/EditableListView.swift (hunk 3 above)
Comment (tendai-moffatt) [reply Sources/SUIComponents/Organisms/EditableListView.swift]: This has been removed now

**Review by paulb00th — APPROVED** [review]

**Pull request merged**